diff --git a/src/main/jhipster/src/main/webapp/app/entities/sample/tree-table/sample-tree.service.ts b/src/main/jhipster/src/main/webapp/app/entities/sample/tree-table/sample-tree.service.ts
index a4c50d7ff..948083cb7 100644
--- a/src/main/jhipster/src/main/webapp/app/entities/sample/tree-table/sample-tree.service.ts
+++ b/src/main/jhipster/src/main/webapp/app/entities/sample/tree-table/sample-tree.service.ts
@@ -85,4 +85,8 @@ export class SampleTreeService implements TreeService {
     init() {
         return this.expand('');
     }
+
+    persist(folders: string[]) {
+        // TODO
+    }
 }
diff --git a/src/main/jhipster/src/main/webapp/app/shared/tree/germplasm/germplasm-tree.service.ts b/src/main/jhipster/src/main/webapp/app/shared/tree/germplasm/germplasm-tree.service.ts
index ea1663030..85d1a3c3b 100644
--- a/src/main/jhipster/src/main/webapp/app/shared/tree/germplasm/germplasm-tree.service.ts
+++ b/src/main/jhipster/src/main/webapp/app/shared/tree/germplasm/germplasm-tree.service.ts
@@ -5,7 +5,6 @@ import { map } from 'rxjs/operators';
 import { Injectable } from '@angular/core';
 import { ParamContext } from '../../service/param.context';
 import { Observable } from 'rxjs';
-import { Sample } from '../../../entities/sample';
 
 @Injectable()
 export class GermplasmTreeService extends TreeService {
@@ -95,6 +94,19 @@ export class GermplasmTreeService extends TreeService {
         return this.http.put<HttpResponse<number>>(url, { observe: 'response' }, {params});
     }
 
+    persist(foldersParam: string[]): any {
+
+        const url = `${this.resourceUrl}/germplasm-lists/user-tree`;
+        const body = {
+            programUUID: this.paramContext.programUUID,
+            userId: this.paramContext.loggedInUserId,
+            folders: foldersParam
+        };
+        console.log('persist ' + url);
+        console.log(body);
+        return this.http.post(url, body);
+    }
+
     private toTreeNode(item: any, parentKey: any): TreeNode {
         const treeNode = <TreeNode>({
             name: item.title,
diff --git a/src/main/jhipster/src/main/webapp/app/shared/tree/study/study-tree.service.ts b/src/main/jhipster/src/main/webapp/app/shared/tree/study/study-tree.service.ts
index 553f7829e..e1a7b4eb8 100644
--- a/src/main/jhipster/src/main/webapp/app/shared/tree/study/study-tree.service.ts
+++ b/src/main/jhipster/src/main/webapp/app/shared/tree/study/study-tree.service.ts
@@ -45,7 +45,24 @@ export class StudyTreeService extends TreeService {
     }
 
     init() {
-        return this.expand('');
+        const url = `/bmsapi/crops/${this.paramContext.cropName}/studies/user-tree`;
+        const params = {
+            onlyFolders: '0'
+        };
+        if (this.paramContext.loggedInUserId) {
+            params['userId'] = this.paramContext.loggedInUserId;
+        }
+        if (this.paramContext.programUUID) {
+            params['programUUID'] = this.paramContext.programUUID;
+        }
+        return this.http.get<TreeNode[]>(url, {
+            params,
+            observe: 'response'
+        }).pipe(map((res: any) => res.body.map((item) => this.toTreeNode(item, null))));
+    }
+
+    persist(folders: string[]) {
+        // TODO
     }
 
     private toTreeNode(item: any, parentKey: any): TreeNode {
diff --git a/src/main/jhipster/src/main/webapp/app/shared/tree/tree.component.ts b/src/main/jhipster/src/main/webapp/app/shared/tree/tree.component.ts
index 395c2c7de..725c09d53 100644
--- a/src/main/jhipster/src/main/webapp/app/shared/tree/tree.component.ts
+++ b/src/main/jhipster/src/main/webapp/app/shared/tree/tree.component.ts
@@ -10,6 +10,7 @@ export class TreeComponent implements OnInit {
 
     public nodes: PrimeNgTreeNode[] = [];
     selectedNodes: PrimeNgTreeNode[];
+    expandedNodes: PrimeNgTreeNode[] = [];
 
     constructor(public service: TreeService,
                 public activeModal: NgbActiveModal) {
@@ -46,23 +47,45 @@ export class TreeComponent implements OnInit {
         const node: PrimeNgTreeNode = event.node;
         if (!node.expanded) {
             this.expand(node);
-            node.expanded = true;
         }
     }
 
     selectLists() {
-        const selected = this.selectedNodes.filter((node: PrimeNgTreeNode) => node.leaf)
+        const persistPromise = this.persistTreeState();
+        persistPromise.then(() => {
+            console.log('selected nodes ')
+            const selected = this.selectedNodes.filter((node: PrimeNgTreeNode) => node.leaf)
+                .map((node: PrimeNgTreeNode) => {
+                    return {
+                        id: node.data.id,
+                        name: node.data.name
+                    };
+                });
+            this.activeModal.close(selected);
+        });
+    }
+
+    private async persistTreeState() {
+        const expanded = this.expandedNodes
             .map((node: PrimeNgTreeNode) => {
-                return {
-                    id: node.data.id,
-                    name: node.data.name
-                };
+                return node.data.id;
             });
-        this.activeModal.close(selected);
+
+        await this.service.persist(expanded).subscribe();
+        return Promise.resolve()
     }
 
     closeModal() {
-        this.activeModal.dismiss();
+        const persistPromise = this.persistTreeState();
+        persistPromise.then(() => {
+            this.activeModal.dismiss();
+        });
+    }
+
+    private setAsEpanded(node: PrimeNgTreeNode) {
+        console.log('expanding ' + node.data.id)
+        node.expanded = true;
+        this.expandedNodes.push(node);
     }
 
     expand(parent) {
@@ -87,7 +110,7 @@ export class TreeComponent implements OnInit {
                     this.addChildren(child, node.children)
                 }
             });
-            parent.expanded = true;
+           this.setAsEpanded(parent);
         }
     }
 
diff --git a/src/main/jhipster/src/main/webapp/app/shared/tree/tree.service.ts b/src/main/jhipster/src/main/webapp/app/shared/tree/tree.service.ts
index c78ab08a9..452304dc2 100644
--- a/src/main/jhipster/src/main/webapp/app/shared/tree/tree.service.ts
+++ b/src/main/jhipster/src/main/webapp/app/shared/tree/tree.service.ts
@@ -5,4 +5,5 @@ export abstract class TreeService {
     abstract delete(folderId: string): any;
     abstract move(source: string, target: string, isParentCropList: boolean): any;
     abstract init(): any;
+    abstract persist(folders: string[]): any;
 }
